map/ViewController.swift

```diff
@@ -22,22 +22,53 @@ class ViewController: UIViewController {
     @IBAction func saveFileButtonTouchUp(_ sender: UIButton) {
         if let title = saveFileButton.titleLabel?.text {
             if title == "SAVE FILE" {
-                SaveFileTest.share.saveFile()
-                showAlert(title: "File save", message: "File saved. Go to check it out.")
-                saveFileButton.setTitle("CHECK FILE", for: UIControlState.normal)
+                showAlert(title: "Where you want to put in", message: "Select where the file will be put into. Do you want the file will be shown on \"Files\" app?", type: AlertType.CHOICE, actionUpTitle: "Shown", actionDownTitle: "Not Shown")
             } else {
                 showAlert(title: "File read", message: SaveFileTest.share.readFile())
                 saveFileButton.setTitle("SAVE FILE", for: UIControlState.normal)
             }
         }
     }
     
-    func showAlert(title _title: String, message _message: String) {
+    func showAlert(title _title: String, message _message: String, type _type: AlertType = AlertType.INFO, actionUpTitle upTitle: String = "", actionDownTitle downTitle: String = "") {
         let alert = UIAlertController(title: _title, message: _message, preferredStyle: .alert)
-        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+        
+        switch (_type) {
+        case AlertType.INFO:
+            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+            break;
+        case AlertType.CHOICE:
+            if !upTitle.isEmpty && !downTitle.isEmpty {
+                alert.addAction(UIAlertAction(title: upTitle, style: .default,
+                    handler: { action in
+                        DispatchQueue.main.async {
+                            self.share(url: SaveFileTest.share.fileShareURL)
+                        }
+                }))
+                alert.addAction(UIAlertAction(title: downTitle, style: .default,
+                    handler: { action in
+                        SaveFileTest.share.saveFile()
```

Comment:
this line is look like a point. other alert method lines are looks just tutorial things.

Reply:
Yes it seems but it is written only for "development", not further used you know :)